garak/command.py

```diff
@@ -147,16 +152,46 @@ def end_run():
     logging.info(msg)
 
 
-def print_plugins(prefix: str, color):
-    from colorama import Style
+def print_plugins(prefix: str, color, selected_plugins=None):
+    """
+    Print plugins for a category (probes/detectors/generators/buffs).
 
+    Args:
+        prefix: Plugin category (probes/detectors/generators/buffs)
+        color: Color for output formatting
+        selected_plugins: Optional list of specific plugins to show. If None, shows all.
+    """
+    from colorama import Style
     from garak._plugins import enumerate_plugins
 
-    plugin_names = enumerate_plugins(category=prefix)
-    plugin_names = [(p.replace(f"{prefix}.", ""), a) for p, a in plugin_names]
-    module_names = set([(m.split(".")[0], True) for m, a in plugin_names])
-    plugin_names += module_names
-    for plugin_name, active in sorted(plugin_names):
+    # enumerate with activation flags
+    rows = enumerate_plugins(
+        category=prefix
+    )  # [("probes.dan.AntiDAN", active_bool), ...]
+    short = [(p.replace(f"{prefix}.", ""), a) for p, a in rows]
+
+    # Apply filtering if specified
+    if selected_plugins is not None:
+        # Convert selected_plugins to the same format (remove prefix)
+        filtered_short_names = [
+            p.replace(f"{prefix}.", "")
+            for p in selected_plugins
+            if p.startswith(f"{prefix}.")
+        ]
+        short = [(n, a) for (n, a) in short if n in filtered_short_names]
+
+        if not short:
+            print(f"No {prefix} match the provided filter")
+            sys.exit(2)
+
+    # Add family headers for any families that have members
+    module_names = {n.split(".")[0] for n, _ in short}
+    short += [
+        (m, True) for m in module_names
+    ]  # family lines (always active for banner)
```

Comment:
I see the note here, however I am not seeing the value in having a `family` entry when passed a `selected_plugins` set. I suspect this is due to the comment from @leondz however this not the desired function.

The original issue noted the goal here is to get the list of plugin classes that will be used a runtime, a `family` is not `used` it can be provided in the spec to denote use of all `active` classes in that family.

The examples in the issue were intentional though maybe a bit unclear to other reviewers.